src/NLog/Targets/ColoredConsoleTarget.cs

```diff
@@ -177,14 +185,31 @@ protected override void CloseTarget()
             base.CloseTarget();
         }
 
-            /// <summary>
+        /// <summary>
         /// Writes the specified log event to the console highlighting entries
         /// and words based on a set of defined rules.
         /// </summary>
         /// <param name="logEvent">Log event.</param>
         protected override void Write(LogEventInfo logEvent)
         {
-            this.Output(logEvent, this.Layout.Render(logEvent));
+            if (!FastModus)
+            {
+                this.Output(logEvent, this.Layout.Render(logEvent));
+            }
+            else
+            {
+                this.FastOutput(logEvent, this.Layout.Render(logEvent));
+            }
+
+        }
+
+        private void FastOutput(LogEventInfo logEvent, string render)
+        {
+
+            ConsoleRowHighlightingRule matchingRule = FindConsoleRowHighlightingRule(logEvent);
```

Comment:
The underline ConsoleRowHighlightingRule is still using regex, however, this particular method just finds the rule, but is not rendering anything.

The fast modus just use the color definition of the highlighting rule.